src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -208,7 +208,7 @@ public final Connection getConnection(final long hardTimeout) throws SQLExceptio
       if (originalException instanceof SQLException) {
          sqlState = ((SQLException) originalException).getSQLState();
       }
-      throw new SQLTimeoutException(String.format("Timeout after %dms of waiting for a connection.", clockSource.elapsedMillis(startTime)), sqlState, originalException);
+      throw new SQLTimeoutException(poolName + " - Cannot acquire connection, Timeout after " + clockSource.elapsedMillis(startTime) + "ms.", sqlState, originalException);
```

Comment:
Formatting much better, btw.  I'd like different wording than "Cannot acquire connection" ... it is ambiguous and the user might think it implies an acquire from the DB directly.  A little longer message is acceptable, "No connection available, ...".  Also, please de-capitalize "Timeout".